configx/provider.go

```diff
@@ -235,14 +235,13 @@ func (p *Provider) addConfigFile(ctx context.Context, path string, k *koanf.Koan
 				cancel()
 				cancel = cancelInner
 				p.runOnChanges(e, nil)
-				close(c)
```

Comment:
This caused a panic in some new test case because the receiver closed the channel instead of the sender.